### PR TITLE
Add a TimeField class to Pyface

### PR DESCRIPTION
**Pull request by corranwebster**
This adds a time editing field to the `pyface.fields` API.  This is a simple field with no bounds checking (Qt supports this Wx doesn't without significant additional work).

The ulterior motive for this is to see if this can be the basis for the TraitsUI `TimeEditor`, allowing us to move all the toolkit-specific code down to Pyface.  In the meantime, this will allow adding time editors to toolbars.

### CODE REVIEW

File: pyface/fields/i_time_field.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/fields/i_time_field.py:1]:
```suggestion
# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
```

File: pyface/fields/tests/test_time_field.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/fields/tests/test_time_field.py:1]:
```suggestion
# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
```

File: pyface/fields/time_field.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/fields/time_field.py:1]:
```suggestion
# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
```

File: pyface/ui/qt4/fields/time_field.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/ui/qt4/fields/time_field.py:1]:
```suggestion
# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
```

File: pyface/ui/qt4/util/datetime.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/ui/qt4/util/datetime.py:1]:
```suggestion
# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
```

File: pyface/ui/qt4/util/tests/test_datetime.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/ui/qt4/util/tests/test_datetime.py:1]:
```suggestion
# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
```

File: pyface/ui/wx/fields/time_field.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/ui/wx/fields/time_field.py:1]:
```suggestion
# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
```

File: pyface/ui/qt4/fields/time_field.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+#
+# Author: Enthought, Inc.
+# Description: <Enthought pyface package component>
```
Comment (rahulporuri) [inline pyface/ui/qt4/fields/time_field.py:12]:
```suggestion
```

File: pyface/ui/qt4/fields/time_field.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+""" The Qt-specific implementation of the time field class """
+
+
+from traits.api import provides
+
+from pyface.fields.i_time_field import ITimeField, MTimeField
+from pyface.qt.QtGui import QTimeEdit
```
Comment (rahulporuri) [inline pyface/ui/qt4/fields/time_field.py:20]: Following the convention in other parts of pyface, do you think it'll be better to move this import before the `traits.api` import?

File: pyface/ui/qt4/util/datetime.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+#
+# Author: Enthought, Inc.
+# Description: <Enthought pyface package component>
```
Comment (rahulporuri) [inline pyface/ui/qt4/util/datetime.py:12]:
```suggestion
```

File: pyface/ui/qt4/util/datetime.py [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        The corresponding datetime.time.
+    """
+    if is_pyqt:
+        return qtime.toPyTime()
+    else:
+        return qtime.toPython()
+
+
+def pytime_to_qtime(pytime):
+    """ Converta Python datetime.time to a QTime
```
Comment (rahulporuri) [inline pyface/ui/qt4/util/datetime.py:42]:
```suggestion
    """ Convert a Python datetime.time to a QTime
```

File: pyface/ui/qt4/util/tests/test_datetime.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+#
+# Author: Enthought, Inc.
+# Description: <Enthought pyface package component>
```
Comment (rahulporuri) [inline pyface/ui/qt4/util/tests/test_datetime.py:12]:
```suggestion
```

File: pyface/ui/wx/fields/time_field.py [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+#
+# Author: Enthought, Inc.
+# Description: <Enthought pyface package component>
```
Comment (rahulporuri) [inline pyface/ui/wx/fields/time_field.py:12]:
```suggestion
```

**Review by rahulporuri — APPROVED** [review]
LGTM with a bunch of mostly nitpicky comments.

File: pyface/ui/qt4/fields/time_field.py (hunk 9 above)
Comment (corranwebster) [reply pyface/ui/qt4/fields/time_field.py:20]: Oops, yes

**Pull request merged**